### PR TITLE
Add LinearLayout

### PR DESCRIPTION
**Pull request by beansgum**
LinearLayout serves as a wrapper for several gio layout structs and it reduces the amount of code required when applying basic attributes like padding, margin, border, etc to a layout. The number of lines of code used for the most basic layouts can sometimes look very complex due to the multiple nested functions.

This layout code:
```go
return layout.Inset{Top: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
	return layout.Stack{}.Layout(gtx,
		layout.Expanded(func(gtx C) D {
			// // draw background and and clip the background to border radius
			tr := float32(gtx.Px(unit.Dp(Border.Radius.TopRight)))
			tl := float32(gtx.Px(unit.Dp(Border.Radius.TopLeft)))
			br := float32(gtx.Px(unit.Dp(Border.Radius.BottomRight)))
			bl := float32(gtx.Px(unit.Dp(Border.Radius.BottomLeft)))
			clip.RRect{
				Rect: f32.Rectangle{Max: f32.Point{
					X: float32(gtx.Constraints.Min.X),
					Y: float32(gtx.Constraints.Min.Y),
				}},
				NW: tl, NE: tr, SE: br, SW: bl,
			}.Add(gtx.Ops)
			return fill(gtx, bgColor)
		}),
		layout.Stacked(func(gtx C) D {
			return Border{Color: borderColor, Width: values.MarginPadding2}.Layout(gtx, func(gtx C) D {
				return layout.Inset{Top: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
					return layout.E.Layout(gtx, func(gtx C) D {
						return layout.Flex{Axis: layout.Vertical}.Layout(gtx, children...)
					})
				})
			})
		}),
	)
})
```
Can be replaced by this using the LinearLayout:
```
decredmaterial.LinearLayout{
	Width:       decredmaterial.MATCH_PARENT,
	Height:      decredmaterial.WRAP_CONTENT,
	Orientation: layout.Vertical,
	Background:  pg.Theme.Color.Orange,
	Margin:      layout.Inset{Top: values.MarginPadding16},
	Padding:     layout.UniformInset(values.MarginPadding16),
	Border:      decredmaterial.Border{Width: values.MarginPadding1, Color: pg.Theme.Color.Danger},
	Direction:   layout.Center,
}
```
I created an example of LinearLayout [here](https://pastebin.com/Tsb7gVq7) and this is the result:

### CODE REVIEW

**Comment by Sirmorrison** [comment]
image

clicking on the receive page, this is the UI that shows up.
I believe you where testing out what you implemented. 
This can be done on the testapp package.

File: ui/page/more_page.go [hunk 1]
```diff
@@ -108,47 +107,129 @@ func (pg *MorePage) Layout(gtx layout.Context) layout.Dimensions {
...
+				Border:     decredmaterial.Border{Radius: decredmaterial.Radius(10)},
+			}.Layout(gtx, layout.Rigid(pg.Theme.Body1("Radius & Padding").Layout))
+		}),
+		layout.Rigid(func(gtx C) D {
+			return decredmaterial.LinearLayout{
+				Width:      decredmaterial.MatchParent,
+				Height:     decredmaterial.MatchParent,
+				Margin:     layout.Inset{Top: values.MarginPadding16},
+				Background: pg.Theme.Color.Orange,
+			}.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/more_page.go]: move test layout codes to testApp package.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

File: ui/decredmaterial/linearlayout.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	Direction   layout.Direction
+}
+
+func (ll LinearLayout) Layout(gtx C, children ...layout.FlexChild) D {
+	// draw margin
+	return ll.Margin.Layout(gtx, func(gtx C) D {
+		return layout.Stack{}.Layout(gtx,
+			layout.Expanded(func(gtx C) D {
+				ll.applyDimension(&gtx)
+				// // draw background and and clip the background to border radius
```
Comment (Sirmorrison) [inline ui/decredmaterial/linearlayout.go]:
```suggestion
				// draw background and clip the background to border radius
```

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
You did not add the sample layout code to the `test` package for code reference.

**Review by Sirmorrison — DISMISSED** [review]

**Review by dreacot — DISMISSED** [review]

**Review by dreacot — APPROVED** [review]

**Review by Sirmorrison — APPROVED** [review]

**Pull request merged**